### DISCUSSION TITLE
Route to new page from function

### DISCUSSION THREAD

**Discussion by iamtetea** [quasarframework/quasar#10144] [CLI - PWA mode]
How to route to new page from funtions?

this.$router.push('/route-name') is not working inside the script function

`  setup () {
      const $q = useQuasar()

      function showBottomSheet () {
        $q.bottomSheet({
          message: 'MANAGEMENTS',
          actions: [
            {
              label: 'Users',
              icon: 'r_groups',
              color: $q.dark.isActive ? 'grey-3' : 'grey-8',
              id: 'users'
            },
            {},
            {
              label: 'LOGOUT',
              icon: 'r_logout',
              color: 'negative',
              id: 'logout'
            }
          ]
        }).onOk(action => {
          if (action.id === 'logout') {
            // logout
          } else {
            this.$router.push('/asd')
          }
        })
      }

      return {
        showBottomSheet
      }
  },`

**Comment by MilosPaunovic** [comment]
Do you get any errors in the console?

**Comment by pdanpdan** [comment]
https://v3.vuejs.org/guide/composition-api-setup.html#usage-of-this

**Comment by mfoitzik** [comment]
I have the same question and I think I am missing something fundamental that I cannot figure out. Inside setup I have a function that I want to use to change the route. I can successfully use window.location.href as shown below. But how do I get access to the router inside functions inside setup?

```
setup() {
    const singleOrCoapp = ref('single')
    function goToTerms() {
      window.location.href = '#/terms'
    }
    return { singleOrCoapp, goToTerms };
  }
```

**Comment by pdanpdan** [comment]
Impressive. No one even bothered to write the text in Google before writing it here.

https://lmgtfy.app/?q=vue3+how+do+I+get+access+to+the+router+inside+functions+inside+setup

**Reply by mfoitzik** [reply]
Thank you pdandan. I had never heard about this site called Google before. I may start using it occasionally to look for answers to my questions. Just in case someone else who is just getting used to the composition API inside of a Quasar app is looking for the same answer and lands here, here is what I did:

```
import { useRouter } from 'vue-router'
export default defineComponent({
  name: 'PageIndex',
  components: { },
  setup() {
    const router = useRouter()
    function goToTerms() {
      void router.push('terms')
    }
    return { goToTerms };
  }
```

**Reply by hawkeye64** [reply]
Why `void router.push('terms')` (void) here -- not needed

**Reply by mfoitzik** [reply]
@hawkeye64 I get an eslint error (_Promises must be handled appropriately or explicitly marked as ignored with the `void` operator.eslint@typescript-eslint/no-floating-promises_) without it. This was my quick fix.
